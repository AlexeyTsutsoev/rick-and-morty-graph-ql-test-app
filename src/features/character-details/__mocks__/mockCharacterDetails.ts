import { GetCharacterByIdQuery } from "../../../services/network/__generated__/graphql";
import { CharacterDetailsRenderData, EpisodeRenderData } from "../types";

export const VALID_MOCK_CHARACTER_DETAILS: GetCharacterByIdQuery = {
  character: {
    name: "Morty Smith",
    image: "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
    species: "Human",
    gender: "Male",
    episode: [
      {
        id: "1",
        name: "Pilot",
        air_date: "December 2, 2013",
      },
      {
        id: "2",
        name: "Lawnmower Dog",
        air_date: "December 9, 2013",
      },
    ],
  },
};

export const INVALID_MOCK_CHARACTER_DETAILS = {
  character: {
    name: "Morty Smith",
    image: "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
    species: "Human",
    episode: [
      {
        id: "1",
        name: "Pilot",
        air_date: "December 2, 2013",
      },
      {
        name: "Lawnmower Dog",
        air_date: "December 9, 2013",
      },
    ],
  },
};

export const INVALID_MOCK_EPISODE = {
  name: "Lawnmower Dog",
  air_date: "December 9, 2013",
};

export const VALID_MOCK_EPISODE = {
  id: "1",
  name: "Pilot",
  air_date: "December 2, 2013",
};

export const VALID_MOCK_EPISODE_RENDER_DATA: EpisodeRenderData = {
  id: "1",
  name: "Pilot",
  date: "December 2, 2013",
};

export const VALID_MOCK_CHARACTER_DETAILS_RENDER_DATA: CharacterDetailsRenderData = {
  name: "Morty Smith",
  image: "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
  species: "Human",
  gender: "Male",
  episodes: [
    {
      id: "1",
      name: "Pilot",
      date: "December 2, 2013",
    },
    {
      id: "2",
      name: "Lawnmower Dog",
      date: "December 9, 2013",
    },
  ],
};
