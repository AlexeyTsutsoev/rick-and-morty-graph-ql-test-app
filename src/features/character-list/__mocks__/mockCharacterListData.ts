import { GetCharacterListQuery } from "../../../services/network/__generated__/graphql";
import { CharacterRenderData } from "../types";

export const VALID_MOCK_CHARACTER_RESPONSE: GetCharacterListQuery = {
  characters: {
    info: {
      count: 826,
      pages: 42,
    },
    results: [
      {
        name: "Rick Sanchez",
        id: "1",
        image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
      },
      {
        name: "Morty Smith",
        id: "2",
        image: "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
      },
    ],
  },
};

export const INVALID_MOCK_CHARACTER_RESPONSE: GetCharacterListQuery = {
  characters: {
    results: [
      {
        id: "1",
        image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
      },
      {
        name: "Morty Smith",
        image: "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
      },
    ],
  },
};

export const INVALID_MOCK_CHARACTER = {
  name: "Morty Smith",
  image: "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
};

export const VALID_MOCK_CHARACTER = {
  name: "Morty Smith",
  id: "2",
  image: "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
};

export const VALID_MOCK_RENDER_DATA: CharacterRenderData[] = [
  {
    name: "Rick Sanchez",
    id: "1",
    image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
  },
  {
    name: "Morty Smith",
    id: "2",
    image: "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
  },
];
