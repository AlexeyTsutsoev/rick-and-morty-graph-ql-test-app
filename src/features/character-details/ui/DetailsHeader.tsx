import * as React from "react";
import { Image, StyleSheet, View } from "react-native";

import { CharacterDetailsRenderData } from "../types";
import { Typography } from "../../../shared/ui/components/atoms";
import { getColor, scaleHeight, scaleWidth } from "../../../shared/ui/theme";

interface DetailsHeaderProps {
  details: CharacterDetailsRenderData;
}

const detailsArray: (keyof CharacterDetailsRenderData)[] = ["gender", "species"];

export const DetailsHeader: React.FC<DetailsHeaderProps> = ({ details }) => {
  return (
    <View style={styles.container}>
      <View style={styles.rowContainer}>
        <Image style={styles.image} source={{ uri: details.image }} />

        <Typography variant="headingLarge">{details.name}</Typography>
      </View>

      {detailsArray.map(detail => {
        return (
          <View style={styles.rowContainer} key={detail}>
            <Typography color="secondary">{detail}: </Typography>

            <Typography>{details[detail]}</Typography>
          </View>
        );
      })}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginBottom: scaleHeight(10),
    borderBottomColor: getColor("accent"),
    borderBottomWidth: 1,
  },
  rowContainer: {
    flexDirection: "row",
    alignItems: "center",
    marginBottom: scaleHeight(10),
  },
  image: {
    width: scaleHeight(80),
    height: scaleHeight(80),

    borderRadius: 50,

    marginRight: scaleWidth(20),
  },
});
