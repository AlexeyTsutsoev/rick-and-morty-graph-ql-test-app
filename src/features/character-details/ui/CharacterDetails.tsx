import * as React from "react";
import { FlatList, StyleSheet, View } from "react-native";

import { CharacterDetailsRenderData } from "../types";
import { ErrorState } from "../../../shared/ui/components/organisms";
import { keyExtractor } from "../../../shared/utils";

import { DetailsHeader } from "./DetailsHeader";
import { EpisodeCard } from "./EpisodeCard";
import { getColor, scaleHeight, scaleWidth } from "../../../shared/ui/theme";
import { DetailsSkeleton } from "./DetailsSkeleton";

interface CharacterDetailsProps {
  data: CharacterDetailsRenderData | null;
  isLoading: boolean;
  isError: boolean;
  onRefetch: () => void;
}

export const CharacterDetails: React.FC<CharacterDetailsProps> = ({ data, isLoading, isError, onRefetch }) => {
  if (isLoading) {
    return (
      <View style={[styles.container, styles.loaderContainer]}>
        <DetailsSkeleton />
      </View>
    );
  }

  if (!data || isError) {
    return <ErrorState onRefetch={onRefetch} />;
  }

  return (
    <FlatList
      contentContainerStyle={[styles.container, styles.contentContainer]}
      keyExtractor={keyExtractor}
      data={data.episodes}
      ListHeaderComponent={<DetailsHeader details={data} />}
      renderItem={({ item }) => <EpisodeCard episode={item} />}
    />
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: getColor("background"),
  },
  contentContainer: {
    paddingHorizontal: scaleWidth(20),
    paddingVertical: scaleHeight(10),
  },
  loaderContainer: {
    flex: 1,
  },
});
