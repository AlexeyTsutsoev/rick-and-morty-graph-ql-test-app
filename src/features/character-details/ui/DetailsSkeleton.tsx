import * as React from "react";

import { Skeleton, SkeletonItem } from "../../../shared/ui/components/organisms";
import { scaleHeight, scaleWidth } from "../../../shared/ui/theme";

const skeletonEpisodes = Array(10).fill(0);

export const DetailsSkeleton: React.FC = () => {
  return (
    <Skeleton>
      <SkeletonItem paddingHorizontal={scaleWidth(20)} paddingTop={scaleHeight(10)}>
        <SkeletonItem flexDirection="row" alignItems="center" marginBottom={scaleHeight(20)}>
          <SkeletonItem
            borderRadius={50}
            marginRight={scaleWidth(20)}
            height={scaleHeight(80)}
            width={scaleHeight(80)}
          />

          <SkeletonItem width={scaleWidth(150)} height={scaleHeight(28)} />
          <SkeletonItem />
        </SkeletonItem>

        <SkeletonItem width={scaleWidth(200)} height={scaleHeight(24)} marginBottom={scaleHeight(10)} />

        <SkeletonItem width={scaleWidth(200)} height={scaleHeight(24)} marginBottom={scaleHeight(10)} />

        <SkeletonItem width="100%" height={1} marginBottom={scaleHeight(10)} />

        {skeletonEpisodes.map((_, index) => {
          return (
            <SkeletonItem key={index.toString()} width="100%" height={scaleHeight(50)} marginBottom={scaleHeight(5)} />
          );
        })}
      </SkeletonItem>
    </Skeleton>
  );
};
