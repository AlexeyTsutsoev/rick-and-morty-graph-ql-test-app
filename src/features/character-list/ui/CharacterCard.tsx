import * as React from "react";
import { Image, StyleSheet } from "react-native";
import { AnimatedPressable, Typography } from "../../../shared/ui/components/atoms";
import { getColor, scaleHeight, scaleWidth } from "../../../shared/ui/theme";
import { CharacterRenderData, CardPressFunc } from "../types";

interface CharacterCardProps {
  character: CharacterRenderData;
  onPress: CardPressFunc;
}

export const CharacterCard: React.FC<CharacterCardProps> = ({ character, onPress }) => {
  const pressHandler = React.useCallback(() => {
    onPress(character.id);
  }, [character.id, onPress]);
  return (
    <AnimatedPressable style={styles.container} onPress={pressHandler}>
      <Image style={styles.image} source={{ uri: character.image }} />

      <Typography variant="headingSmall">{character.name}</Typography>
    </AnimatedPressable>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "flex-start",

    paddingHorizontal: scaleWidth(20),
    paddingVertical: scaleHeight(10),

    backgroundColor: getColor("background"),

    borderColor: getColor("accent"),
    borderWidth: 2,
    borderRadius: 4,
    marginBottom: scaleHeight(10),
  },

  image: {
    width: scaleHeight(36),
    height: scaleHeight(36),

    borderRadius: 50,

    marginRight: scaleWidth(12),
  },
});
