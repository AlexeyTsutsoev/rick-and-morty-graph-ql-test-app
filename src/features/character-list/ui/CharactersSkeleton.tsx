import * as React from "react";

import { Skeleton, SkeletonItem } from "../../../shared/ui/components/organisms";
import { scaleHeight } from "../../../shared/ui/theme";

const array = Array(10).fill(0);

export const CharactersSkeleton: React.FC = () => {
  return (
    <Skeleton>
      <SkeletonItem>
        {array.map((_, index) => {
          return (
            <SkeletonItem key={index.toString()} width="100%" marginBottom={scaleHeight(10)} height={scaleHeight(56)} />
          );
        })}
      </SkeletonItem>
    </Skeleton>
  );
};
