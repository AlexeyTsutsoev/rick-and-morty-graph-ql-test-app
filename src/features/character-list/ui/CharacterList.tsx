import * as React from "react";
import { FlatList, StyleSheet } from "react-native";

import { CharacterRenderData, TCardPressFunc } from "../types";
import { ErrorState } from "../../../shared/ui/components/organisms";
import { CharacterCard } from "./CharacterCard";
import { scaleHeight } from "../../../shared/ui/theme";
import { CharactersSkeleton } from "./CharactersSkeleton";

interface CharacterListProps {
  data: CharacterRenderData[];
  isLoading: boolean;
  isError: boolean;
  increasePage: () => void;
  onRefetch: () => void;
  onCardPress: TCardPressFunc;
}

export const CharacterList: React.FC<CharacterListProps> = ({
  data,
  isLoading,
  isError,
  onCardPress,
  increasePage,
  onRefetch,
}) => {
  if (!data.length && isLoading) {
    // TODO: Add Skeletons
    return <CharactersSkeleton />;
  }

  if (!data.length && isError) {
    return <ErrorState onRefetch={onRefetch} />;
  }

  return (
    <FlatList
      data={data}
      keyExtractor={({ id }) => id}
      contentContainerStyle={styles.contentContainer}
      renderItem={({ item }) => <CharacterCard character={item} onPress={onCardPress} />}
      showsVerticalScrollIndicator={false}
      onEndReached={increasePage}
    />
  );
};

const styles = StyleSheet.create({
  contentContainer: {
    paddingBottom: scaleHeight(20),
  },
});
