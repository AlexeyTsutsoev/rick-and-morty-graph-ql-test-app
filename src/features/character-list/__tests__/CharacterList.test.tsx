import { render } from "@testing-library/react-native";
import React from "react";
import { CharacterList } from "../ui/CharacterList";
import { VALID_MOCK_RENDER_DATA } from "../__mocks__/mockCharacterListData";

describe("Tests for main page", () => {
  it("should match snapshot when default state", () => {
    const component = render(
      <CharacterList
        data={VALID_MOCK_RENDER_DATA}
        isLoading={false}
        isError={false}
        onRefetch={function (): void {
          throw new Error("Function not implemented.");
        }}
        onCardPress={function (): void {
          throw new Error("Function not implemented.");
        }}
        increasePage={function (): void {
          throw new Error("Function not implemented.");
        }}
      />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should match snapshot when loading", () => {
    const component = render(
      <CharacterList
        data={VALID_MOCK_RENDER_DATA}
        isLoading={false}
        isError={false}
        onRefetch={function (): void {
          throw new Error("Function not implemented.");
        }}
        onCardPress={function (): void {
          throw new Error("Function not implemented.");
        }}
        increasePage={function (): void {
          throw new Error("Function not implemented.");
        }}
      />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should match snapshot when data empty and error", () => {
    const component = render(
      <CharacterList
        data={[]}
        isLoading={false}
        isError={true}
        onRefetch={function (): void {
          throw new Error("Function not implemented.");
        }}
        onCardPress={function (): void {
          throw new Error("Function not implemented.");
        }}
        increasePage={function (): void {
          throw new Error("Function not implemented.");
        }}
      />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
