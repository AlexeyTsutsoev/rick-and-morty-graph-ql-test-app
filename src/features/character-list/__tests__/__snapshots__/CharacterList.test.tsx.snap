// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for main page should match snapshot when data empty and error 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#BC0100",
          "fontSize": 39,
          "fontWeight": "800",
        },
        Object {
          "marginBottom": 16,
          "textAlign": "center",
        },
      ]
    }
  >
    Something went wrong, please reload data.
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#BC0100",
        "borderRadius": 16,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 16,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 30,
            "fontWeight": "600",
          },
          undefined,
        ]
      }
    >
      Update
    </Text>
  </View>
</View>
`;

exports[`Tests for main page should match snapshot when default state 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 32,
    }
  }
  data={
    Array [
      Object {
        "id": "1",
        "image": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
        "name": "Rick Sanchez",
      },
      Object {
        "id": "2",
        "image": "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
        "name": "Morty Smith",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#05AFC8",
            "borderRadius": 4,
            "borderWidth": 2,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginBottom": 16,
            "opacity": 1,
            "paddingHorizontal": 40,
            "paddingVertical": 16,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
            }
          }
          style={
            Object {
              "borderRadius": 50,
              "height": 59,
              "marginRight": 24,
              "width": 59,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#121212",
                "fontSize": 36,
                "fontWeight": "600",
              },
              undefined,
            ]
          }
        >
          Rick Sanchez
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#05AFC8",
            "borderRadius": 4,
            "borderWidth": 2,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginBottom": 16,
            "opacity": 1,
            "paddingHorizontal": 40,
            "paddingVertical": 16,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
            }
          }
          style={
            Object {
              "borderRadius": 50,
              "height": 59,
              "marginRight": 24,
              "width": 59,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#121212",
                "fontSize": 36,
                "fontWeight": "600",
              },
              undefined,
            ]
          }
        >
          Morty Smith
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Tests for main page should match snapshot when loading 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 32,
    }
  }
  data={
    Array [
      Object {
        "id": "1",
        "image": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
        "name": "Rick Sanchez",
      },
      Object {
        "id": "2",
        "image": "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
        "name": "Morty Smith",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#05AFC8",
            "borderRadius": 4,
            "borderWidth": 2,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginBottom": 16,
            "opacity": 1,
            "paddingHorizontal": 40,
            "paddingVertical": 16,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
            }
          }
          style={
            Object {
              "borderRadius": 50,
              "height": 59,
              "marginRight": 24,
              "width": 59,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#121212",
                "fontSize": 36,
                "fontWeight": "600",
              },
              undefined,
            ]
          }
        >
          Rick Sanchez
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#05AFC8",
            "borderRadius": 4,
            "borderWidth": 2,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginBottom": 16,
            "opacity": 1,
            "paddingHorizontal": 40,
            "paddingVertical": 16,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
            }
          }
          style={
            Object {
              "borderRadius": 50,
              "height": 59,
              "marginRight": 24,
              "width": 59,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#121212",
                "fontSize": 36,
                "fontWeight": "600",
              },
              undefined,
            ]
          }
        >
          Morty Smith
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
